### PR TITLE
Bug 1520355 - Update docs/testcases.md

### PR DESCRIPTION
**Pull request by KWierso**
This had a few outdated things that should be updated.

### CODE REVIEW

File: docs/testcases.md [hunk 1]
```diff
@@ -194,13 +201,13 @@ Click on another failure line in the failed step.
 
 **Expected**: The log should scroll to that failure line.
 
-Click on 'show successful steps'.
+Click on 'show successful steps'. (DELETE ME? I can't find this on a failing mochitest log)
```
Comment (KWierso) [inline docs/testcases.md]: I'm not sure if these should be removed. Maybe a different job type would show this, or maybe this was removed a long time ago and no longer applies...

**Comment by edmorley** [comment]
Does anyone use this file any more? I have never used it.

If others don't too (and given our push towards more test coverage recently), should we just remove it entirely?

Thoughts, @camd @sarah-clements @ionutgoldan?

**Comment by camd** [comment]
This file was some of the basis for the current Selenium tests.  But I'm not certain which tests here are covered in Selenium.  It might be worth doing an audit there.

I have used this file as a check-list for tests to execute prior to pushing to production in the past.  I haven't for a while now, though.  I could go either way, but my preference would be to keep these till we determine that all the valuable tests are covered by automation.  Or maybe I'm just a hoarder...  :)

File: docs/testcases.md (hunk 1 above)
Comment (camd) [reply docs/testcases.md]: Yeah, this was removed since Taskcluster jobs don't have "steps" like BuildBot jobs did.

**Comment by edmorley** [comment]
Keeping for a bit longer in case any are useful as the basis for Selenium tests sounds fine to me :-)

File: docs/testcases.md [hunk 2]
```diff
@@ -194,13 +201,13 @@ Click on another failure line in the failed step.
...
 
-**Expected**: Green successful step bars should appear in the top right panel.
+**Expected**: Green successful step bars should appear in the top right panel. (DELETE ME? I can't find this on a failing mochitest log)
 
-Click on a successful step.
+Click on a successful step. (DELETE ME? I can't find this on a failing mochitest log)
 
-**Expected**: The log contents should scroll to the -- Start -- line for that step.
+**Expected**: The log contents should scroll to the -- Start -- line for that step. (DELETE ME? I can't find this on a failing mochitest log)
```
Comment (camd) [inline docs/testcases.md]: Yeah, you can remove both of these.  The scrolling should still happen with failure lines, however.

**Comment by sarah-clements** [comment]
I haven't used this file either.

**Comment by camd** [comment]
Looks good.  I'll go ahead and merge it.

**Pull request merged**